pkg/fqdn/cache.go

```diff
@@ -770,6 +782,18 @@ func NewDNSZombieMappings(max, perHostLimit int) *DNSZombieMappings {
 	}
 }
 
+// Names returns the FQDN entries referred to in the DNSZombieMappings
+func (zombies *DNSZombieMappings) Names() []string {
+	zombies.Lock()
+	defer zombies.Unlock()
```

Comment:
```suggestion
	zombies.RLock()
	defer zombies.RUnlock()
```
would suffice, I believe

Reply:
It's not a RWMutex.

Reply:
err you're right - for some reason I thought the DNSZombies were a DNSCache too. I don't think there's much benefit to a RWLock here, given it's per EP and GC is mutating. Let's not.